### PR TITLE
Feature/controllers specs

### PR DESCRIPTION
**Pull request by blessedjasonmwanza**
## Feature/controllers-specs
In this TDD update;
- Configure and use Rspec to write unit tests
- Write unit tests for Users controller to ensure it renders the right content and status
- Write unit tests for Posts controller  to ensure it renders the right content and status

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    end
+    it "response body contains 'Posts'" do
+      get '/users/1/posts'
+      expect(response.body).to include('Posts by user Page')
+    end
+    it 'shows post by id' do
+      get '/users/1/posts/1'
+      expect(response).to have_http_status(200)
+      expect(response.body).to include('Show post by id page')
+    end
```
Comment (Gambit142) [inline spec/requests/posts_spec.rb:26]: - [x] Kindly add a test that shows it renders the template `:show` just like you did with the `:index`

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      expect(response.body).to include('all users')
+    end
+  end
+
+  describe 'GET /users/:id' do
+    it 'shows user info' do
+      get '/users/1'
+      expect(response).to have_http_status(200)
+      expect(response.body).to include('user info')
+    end
```
Comment (Gambit142) [inline spec/requests/users_spec.rb:30]: - [x] Kindly add a test that shows it renders the template `:show` just like you did with the `:index`

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @blessedjasonmwanza 👋 , 

Great job so far, kudos 👏,
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Included `rspec-rails` gem 💯 
- All tests are request tests 💯 

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/requests/users_spec.rb (hunk 2 above)
Comment (blessedjasonmwanza) [reply spec/requests/users_spec.rb:30]: Hello @Gambit142, Thanks a million  for spotting that :-) />

On it.

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (blessedjasonmwanza) [reply spec/requests/posts_spec.rb:26]: Actioned

**Review by bimbolabuari — APPROVED** [review]
Hi @blessedjasonmwanza,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**